test/modal.test.js

```diff
@@ -79,45 +83,46 @@ describe('Modal', () => {
         }}
       />
     )
-    expect(
-      wrapper.find('.modal-fade-screen.custom.modal-is-open').exists()
-    ).toEqual(true)
+    const overlay = screen.getByRole('dialog').parentElement
+    expect(overlay).toHaveClass('modal-fade-screen', 'custom', 'modal-is-open')
   })
 
   describe('when preventClose=true', () => {
     test('hides close button', () => {
-      const wrapper = mount(<Modal preventClose={true} onClose={noop} />)
-      expect(wrapper.find('.modal-close').exists()).toEqual(false)
+      render(<Modal preventClose={true} onClose={noop} />)
+      expect(screen.queryByRole('button', { name: 'Close Modal' })).not.toBeInTheDocument()
     })
 
-    test('does not close by escape key', () => {
+    test('does not close by escape key', async () => {
       const onClose = jest.fn()
-      const wrapper = mount(<Modal preventClose={true} onClose={onClose} />)
-      wrapper.find('.modal-content').simulate('keydown', { keyCode: 27 })
+      const user = userEvent.setup()
+      render(<Modal preventClose={true} onClose={onClose} />)
+
+      await user.keyboard('{Escape}')
       expect(onClose).not.toHaveBeenCalled()
     })
 
-    test('does not close by overlay click', () => {
+    test('does not close by overlay click', async () => {
       const onClose = jest.fn()
-      const overlayClass = 'test-overlay'
-      const wrapper = mount(
+      const user = userEvent.setup()
+      render(
         <Modal
           preventClose={true}
           onClose={onClose}
-          overlayClassName={overlayClass}
         />
       )
-      wrapper.find('.' + overlayClass).simulate('click')
+      const overlay = screen.getByRole('dialog').parentElement
+      await user.click(overlay)
       expect(onClose).not.toHaveBeenCalled()
-      expect(wrapper.find('.modal-content').exists()).toEqual(true)
```

Comment:
This assertion is actually a bit of a false flag – the test wasn't changing `isOpen` state, so even if `onClose` was called, this would still pass. We don't need it.